tests/utils/test_pauli_rep.py

```diff
@@ -420,3 +500,29 @@ def test_symplectic2str(self):
                 ),
             )
         )
+
+    def test_str2str(self):
+        """Tests str2str function"""
+        self.assertEqual(
+            str2str("iX1X3Y4Z9", LATEX_SYNTAX, qubit_order_output="left-to-right"),
+            "iX_{1}X_{3}Y_{4}Z_{9}",
+        )
+        self.assertEqual(
+            str2str("iX_{1}X_{3}Y_{4}Z_{9}", INDEX_SYNTAX, qubit_order_output="left-to-right"),
+            "iX1X3Y4Z9",
+        )
+        self.assertEqual(str2str("IIX", INDEX_SYNTAX), "X0")
+        self.assertEqual(str2str("Y_{3}Y_{4}", INDEX_SYNTAX), "Y4Y3")
+        self.assertEqual(
+            str2str("Y_{3}Y_{4}", PRODUCT_SYNTAX, qubit_order_output="left-to-right"), "IIIYY"
+        )
+        self.assertEqual(
+            str2str(
+                "Y_{3}Y_{4}",
+                PRODUCT_SYNTAX,
+                index_start_input=3,
+                index_start_output=0,
+                qubit_order_output="left-to-right",
+            ),
+            "YY",
+        )
```

Comment:
Can we add a `product -> latex` and `index -> prod` test?

Reply:
Yes, done. When testing product -> latex I realized I needed to add qubit_order_input as an input variable to str2str.

Reply:
Great!